huaweicloud/resource_huaweicloud_bcs_instance_test.go

```diff
@@ -0,0 +1,310 @@
+		}
+		if found.Basic.ID == "" {
+			return fmt.Errorf("resource %s does not exist", name)
+		}
+
+		instance = found
+		return nil
+	}
+}
+
+func testBCSInstanceV2_base(rName string) string {
+	return fmt.Sprintf(`
+data "huaweicloud_availability_zones" "test" {}
+
+resource "huaweicloud_vpc" "test" {
+  name = "%s"
+  cidr = "192.168.0.0/24"
+}
+
+resource "huaweicloud_vpc_subnet" "test" {
+  name          = "%s"
+  cidr          = "192.168.0.0/24"
+  gateway_ip    = "192.168.0.1"
+  primary_dns   = "100.125.1.250"
+  secondary_dns = "100.125.129.250"
+  vpc_id        = huaweicloud_vpc.test.id
+  timeouts {
+    delete = "30m"
+  }
+}
+
+resource "huaweicloud_cce_cluster" "test" {
+  name                   = "%s"
+  flavor_id              = "cce.s2.small"
+  vpc_id                 = huaweicloud_vpc.test.id
+  subnet_id              = huaweicloud_vpc_subnet.test.id
+  container_network_type = "overlay_l2"
+  service_network_cidr   = "10.248.0.0/16"
+  cluster_version        = "v1.15.11-r1"
+  delete_sfs             = true
+}
+
+resource "huaweicloud_cce_node" "test" {
+  name                 = "%s"
+  cluster_id           = huaweicloud_cce_cluster.test.id
+  flavor_id            = "s6.xlarge.2"
+  availability_zone    = data.huaweicloud_availability_zones.test.names[0]
+  password             = "%s"
+  max_pods             = 30
+  ecs_performance_type = "normal"
+  os                   = "CentOS 7.6"
+
+  root_volume {
+    size = 40
+    volumetype = "SAS"
+  }
+  data_volumes {
+    size = 100
+    volumetype = "SAS"
+  }
+}
+`, rName, rName, rName, rName, salt_password)
+}
+
+func testBCSInstanceV2_basic(rName string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_bcs_instance" "test" {
+  depends_on = [huaweicloud_cce_node.test]
+
+  name                  = "%s"
+  version_type          = 1
+  consensus             = "etcdraft"
+  fabric_version        = "2.0"
+  blockchain_type       = "private"
+  enterprise_project_id = "%s"
+  password              = "Huawei@123"
+  cluster_type          = "cce"
+  volume_type           = "nfs"
+  org_disk_size         = 100
+  sign_algorithm        = "ECDSA"
+  orderer_node_num      = 1
+  delete_storage        = true
+
+  cluster_info {
+    id   = huaweicloud_cce_cluster.test.id
+    name = huaweicloud_cce_cluster.test.name
+  }
+  peer_orgs {
+    org_name = "Organization01"
+    channel  = "channeldemo001"
+    count    = 1
+  }
+  channels {
+    name = "channeldemo001"
+  }
+}
+`, testBCSInstanceV2_base(rName), rName, HW_ENTERPRISE_PROJECT_ID_TEST)
+}
+
+func testBCSInstanceV2_kafka(rName string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "huaweicloud_bcs_instance" "test" {
+  depends_on = [huaweicloud_cce_node.test]
+
+  name                  = "%s"
+  version_type          = 4
+  consensus             = "kafka"
+  fabric_version        = "1.4"
+  blockchain_type       = "private"
+  enterprise_project_id = "%s"
+  password              = "Huawei@123"
```

Comment:
can we using `random_password`?

Reply:
I see that the acc test does not support random providers, the message is:
```
        - Could not satisfy plugin requirements: 
        Plugin reinitialization required. Please run "terraform init".
        
        Plugins are external binaries that Terraform uses to access and manipulate
        resources. The configuration provided requires plugins which can't be located,
        don't satisfy the version constraints, or are otherwise incompatible.
        
        Terraform automatically discovers provider requirements from your
        configuration, including providers used in child modules. To see the
        requirements and constraints from each module, run "terraform providers".
        
        - provider "random" is not available
```

Reply:
The configuration of random password is :
```
resource "random_password" "instance_password" {
	length           = 12
	special          = true
	min_lower        = 1
	min_upper        = 1
	override_special = "!@$%%^-_=+[{}]:,./?"
}
```